tests/cmds/cmd_test_utils.py

```diff
@@ -0,0 +1,213 @@
+    fingerprint: int = field(init=False, default=0)
+
+
+@dataclass
+class TestFullNodeRpcClient(TestRpcClient):
+    client_type: Type[FullNodeRpcClient] = field(init=False, default=FullNodeRpcClient)
+
+    async def get_blockchain_state(self) -> Dict[str, Any]:
+        response: Dict[str, Any] = {
+            "peak": cast(BlockRecord, create_test_block_record()),
+            "genesis_challenge_initialized": True,
+            "sync": {
+                "sync_mode": False,
+                "synced": True,
+                "sync_tip_height": 0,
+                "sync_progress_height": 0,
+            },
+            "difficulty": 1024,
+            "sub_slot_iters": 147849216,
+            "space": 29569289860555554816,
+            "mempool_size": 3,
+            "mempool_cost": 88304083,
+            "mempool_fees": 50,
+            "mempool_min_fees": {
+                # We may give estimates for varying costs in the future
+                # This Dict sets us up for that in the future
+                "cost_5000000": 0,
+            },
+            "mempool_max_total_cost": 550000000000,
+            "block_max_cost": DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM,
+            "node_id": "7991a584ae4784ab7525bda352ea9b155ce2ac108d361afc13d5964a0f33fa6d",
+        }
+        self.add_to_log("get_blockchain_state", ())
+        return response
+
+    async def get_block_record_by_height(self, height: int) -> Optional[BlockRecord]:
+        self.add_to_log("get_block_record_by_height", (height,))
+        return cast(BlockRecord, create_test_block_record(height=uint32(height)))
+
+    async def get_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
+        self.add_to_log("get_block_record", (header_hash,))
+        return cast(BlockRecord, create_test_block_record(header_hash=header_hash))
+
+
+@dataclass
+class TestDataLayerRpcClient(TestRpcClient):
+    client_type: Type[DataLayerRpcClient] = field(init=False, default=DataLayerRpcClient)
+
+
+@dataclass
+class TestSimulatorFullNodeRpcClient(TestRpcClient):
+    client_type: Type[SimulatorFullNodeRpcClient] = field(init=False, default=SimulatorFullNodeRpcClient)
+
+
+@dataclass
+class TestRpcClients:
+    """
+    Because this data is in a class, it can be modified by the tests even after the generator is created and imported.
+    """
+
+    farmer_rpc_client: TestFarmerRpcClient = TestFarmerRpcClient()
+    wallet_rpc_client: TestWalletRpcClient = TestWalletRpcClient()
+    full_node_rpc_client: TestFullNodeRpcClient = TestFullNodeRpcClient()
+    data_layer_rpc_client: TestDataLayerRpcClient = TestDataLayerRpcClient()
+    simulator_full_node_rpc_client: TestSimulatorFullNodeRpcClient = TestSimulatorFullNodeRpcClient()
+
+    def get_client(self, client_type: Type[_T_RpcClient]) -> _T_RpcClient:
+        if client_type == FarmerRpcClient:
+            return cast(FarmerRpcClient, self.farmer_rpc_client)  # type: ignore[return-value]
```

Comment:
it looks like you've gone to some length to get the correct type here, with `_T_RpcClient`, why do you still need to suppress this type warning?

Reply:
because its returning a child class, its mypy being stupid.